src/devices/cpu/sh/sh7021.cpp

```diff
@@ -0,0 +1,162 @@
+// license:BSD-3-Clause
+// copyright-holders:Juergen Buchmueller, R. Belmont, Angelo Salese, ...
+
+#include "sh7021.h"
+
+DEFINE_DEVICE_TYPE(SH2A_SH7021, sh2a_sh7021_device, "sh2a_sh7021", "Hitachi SH-2A (SH7021)")
+
+
+sh2a_sh7021_device::sh2a_sh7021_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+	: sh2_device(mconfig, SH2A_SH7021, tag, owner, clock, CPU_TYPE_SH2, address_map_constructor(FUNC(sh2a_sh7021_device::sh7021_map), this), 28, 0xc7ffffff)
+{
+}
+
+void sh2a_sh7021_device::device_start()
+{
+	sh2_device::device_start();
+
+	save_item(NAME(m_sh7021_regs));
+	save_item(NAME(m_dmaor));
+	save_item(STRUCT_MEMBER(m_dma, sar));
+	save_item(STRUCT_MEMBER(m_dma, dar));
+	save_item(STRUCT_MEMBER(m_dma, tcr));
+	save_item(STRUCT_MEMBER(m_dma, chcr));
+}
+
+void sh2a_sh7021_device::device_reset()
+{
+	sh2_device::device_reset();
+
+	std::fill(std::begin(m_sh7021_regs), std::end(m_sh7021_regs), 0);
+
+	m_dmaor = 0;
+
+	for (int i = 0; i < 4; i++) {
+		m_dma[i].sar = 0;
+		m_dma[i].dar = 0;
+		m_dma[i].tcr = 0;
+		m_dma[i].chcr = 0;
+	}
```

Comment:
Looks like K&R sneaking into an Allman file?

Reply:
I threw in some additional save state stuff and initialization in some of these split off CPUs so that's probably my fault. 😇